[Boo Wen Jun] iP

wj200 opened the pull request:
# Mars User Guide

Introducing *MARS*! A personal tasklist chatbot to help you better organise your day

![UI Image](../src/main/docs/Ui.png)

Ever get overwhelmed with Deadlines, Events, ToDos, and DoWithins?   
Here is an assistant with a minimalist UI for you! 😄

# Sneak Peek
- [X] Adding Deadlines  
- [X] Marking Tasks  
- [ ] Reminders *coming soon*

## Get started
Download the .jar file of your desired version.  
Double-click and open file.  
The application is ready for use! Start adding tasks and performing other actions.  

## Here are the types of Tasks available:
* Todo
* Deadline
* Event
* Do Within *a certain time period*

## To add a Todo:
format: todo  {task name}  
*e.g. todo read book*

## To add a deadline:
format: deadline  {task name}  (by: MMM dd yyyy, h:mm a)  
*e.g. deadline return book (by: Feb 02 2024, 6:00 pm)*

## To add an Event:
format: event  {task name}  (from: MMM dd yyyy, h:mm a to MMM dd yyyy, h:mm a)  
*e.g. event CS2103T project meeting (from: Feb 28 2024, 6:00 am to: Mar 03 2024, 8:00 am)*

## To add a DoWithin, run:
format: do within  {task name}  between: MMM dd yyyy, h:mm a and MMM dd yyyy, h:mm a  
*e.g. collect certificate between: Jan 15 2024, 6:00 pm and Jan 25 2024, 6:00 pm*

## If tasks are added successfully, the output would be:

> Got it. I've added this task:  
> [T][ ] read book  
> Now you have 1 task in the list  

> Got it. I've added this task:  
> [D][ ] deadline return book by 2024-02-02 18:00  
> Now you have 2 tasks in the list  

> Got it. I've added this task:  
> [E][ ] CS2103T project meeting from: 2024-02-28 06:00 to 2024-03-03 08:00  
> Now you have 3 tasks in the list  

> Got it. I've added this task:  
> [DW][ ] collect certificate between: 2024-01-15 18:00 and 2024-01-25 18:00  
> Now you have 4 tasks in the list  

## Features:

### Deleting a task
Format: delete {task number} 
*eg. delete 1*
> Noted. I have removed this task [T][ ] read book  
> Now you have 3 tasks in the list

### Marking a task
Format: mark {task number} 
*eg. mark 1*
> Nice! I've marked this task as done: [T][X] read book   

### Unmarking a task
Format: unmark {task number} 
*eg. unmark 1*
> OK, I've marked this task as not done yet: [T][ ] read book    

### List
Format: list 
*e.g. list*
> Here are the tasks in your list:  
> [T][ ] read book  
> [DW][ ] collect certificate between: 2024-01-15 18:00 and 2024-01-25 18:00    

### Find a task through keyword
Format: find {keyword} 
*eg. find book*
> Here are the matching tasks in your list:  
> [T][ ] read book

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

wj200 committed:
Rename,Greet,Exit

wj200 committed:
Level 1. Improve the skeletal version of Duke

wj200 committed:
Level 1. Improve the skeletal version of Duke

wj200 committed:
Level 2.Add, List

wj200 committed:
Level 2.Add, List

wj200 committed:
Level 2.Add, List

wj200 committed:
Level 3. Mark as Done

wj200 committed:
Level 4. ToDos, Events, Deadlines

wj200 committed:
Level 5. Handle Errors

wj200 committed:
Level 6. Delete

wj200 committed:
A-TextUiTesting

wj200 committed:
refactor - remove all usage of arrays from list(), mark(), unmark()

dnardnar commented on src/main/java/marsException.java:
```diff
@@ -0,0 +1,5 @@
+public class marsException extends RuntimeException{
+    public marsException(String message){
```
```suggestion
public class MarsException extends RuntimeException{
    public MarsException(String message){
```
I noticed that the class name marsException does not follow the PascalCase convention for class names. According to the coding standard, class names should be nouns and written in PascalCase. Consider renaming it to MarsException to align with the standard.

dnardnar commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
```
"I like how you used a boolean variable name like isDone, which follows the convention of starting with 'is' to indicate a boolean value. This makes the code more readable and intuitive. Good job!

dnardnar commented on src/main/java/Mars.java:
```diff
@@ -0,0 +1,205 @@
+
+public class Mars {
+    public static void main(String[] args) {
+        List<Task> arrayList = new ArrayList<>();
```
```suggestion
        List<Task> taskList = new ArrayList<>();
```
The variable name arrayList could be improved to better describe its purpose. Since this is a collection of tasks, consider renaming it to tasks or taskList.

dnardnar commented on src/main/java/Mars.java:
```diff
@@ -0,0 +1,205 @@
+    }
+
+    /*Level 2. Add, List */
+    private void store(List<Task> arrayList){
```
The method name store() is a bit ambiguous as it doesn't clearly explain what it does. Based on its functionality, a more descriptive name like addAndListTasks() or manageTasks() would better convey its purpose and improve code clarity

dnardnar reviewed (commented):
I only looked at Code Quality → Naming, this is from week 4, W4.6, take note of this to improve your code and use it in the future as well. Well Done!

wj200 committed:
Abstracted horizontal line into a string constant to improve code readability

wj200 committed:
Created a new Storage class. Abstracted Storage class and Task (and its children) class into relevant packages

wj200 committed:
Added a simple implementation of a static bye() and greet() method to improve code readability

Changed echo() to static method

wj200 committed:
Changed the method signatures of mark() and unmark(). Changed them into static implementations as well.

yuto1115 commented on src/main/java/Mars.java:
```diff
@@ -0,0 +1,205 @@
+                "____________________________________________________________\n" +
+                " Bye. Hope to see you again soon!\n" +
+                "____________________________________________________________"); */
+        try{
```
```suggestion
        try {
```

A space should be inserted here.

yuto1115 commented on src/main/java/Mars.java:
```diff
@@ -0,0 +1,205 @@
+            mars.deleteTask(arrayList);
+        } catch (marsException e) {
+            System.out.println(e.getMessage());
+        }
```
I think these lines can be removed.

yuto1115 commented on src/main/java/Mars.java:
```diff
@@ -0,0 +1,205 @@
+        while(!n.equals("bye")){
+            if(n.equals("list")){
+                Mars.list(arrayList);
+            }
+            else {
```
```suggestion
            } else {
```

Should be modified to follow the coding conventions.

yuto1115 commented on src/main/java/Mars.java:
```diff
@@ -0,0 +1,205 @@
+        String description = reader.nextLine();
+        System.out.println("____________________________________________________________\n");
+        switch(taskType){
+            case "todo":
```
There should not be any indent before "case" statements.

yuto1115 commented on src/main/java/Mars.java:
```diff
@@ -0,0 +1,205 @@
+    private void mark(List<Task> lst){
+        Object[] taskArray = lst.toArray();
+
+        Scanner reader = new Scanner(System.in);
```
Don't you think that the creation of a new `Object[]` is redundant here?

My suggestion is to remove this line and get the task object directly from the given `List<task>` as `lst.get(num - 1)`.

yuto1115 reviewed (commented):
Overall, I found your code fine and easy to understand. You can improve your codes by following the coding style and splitting codes into multiple classes to engage more OOP.

wj200 committed:
Updated method signatures of addTask, deleteTask, list. Standardizing data structure to be ArrayList across all methods

wj200 committed:
Added UI class organized into package. Refactoring of Mars.java. Abstraction of bye() and greet() methods into UI class.

wj200 committed:
Import Java datetime packages

wj200 committed:
Major updates to subclasses of Command. Removed the need for HORIZONTAL_LINE constant at the start of execute() as that functionality has been abstracted into UI class

wj200 committed:
Command, Parser, Storage, Task, and UI packages

wj200 committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091' into A-Gradle

wj200 committed:
Add gradle support to an ongoing project

wj200 committed:
Add JUnit tests to test the behaviour of the code

wj200 committed:
Add JAR file. The task shadowJar creates a JAR file with the name Mars.java and puts it in the build/libs folder

wj200 committed:
Add findCommand class with method implementations. Add
getDescription() method to Task

wj200 committed:
Merge pull request #2 from wj200/A-Jar

A jar

wj200 committed:
Rewrite Storage() constructor to elegantly handle the exception case where file exists (or is successfully created) but opening the file still fails.

wj200 committed:
Added DialogBox, Launcher, Main, and MainWindow classes

wj200 committed:
Updated DialogBox class to handle user inputs

wj200 committed:
Added MainWindow and DialogBox fxml files. Trim DialogBox.java to contain only the controller aspect. Trim Main Class

wj200 committed:
Merge pull request #3 from wj200/Level-10

Merge to master

wj200 committed:
Add Assertions

Delete, mark, unmark classes: add assertions

Subclasses of parent Abstract Command class (e.g. delete, mark, unmark etc.) retrieves a Task instance and stores it into a local variable

Let's assert the condition that the task instance retrieved is not a NULL object

wj200 committed:
Command classes: remove unused import statements

Command child classes (e.g. addCommand, byeCommand, deleteCommand) have some unused import statements

These unused import statements across Command classes cause unnecessary extra lines of code.

Removing these unused import statements helps improve code readability, and therefore code quality.

Let's remove the unused import statements in this classes

wj200 committed:
Merge pull request #4 from wj200/A-Assertions

A-Assertions

wj200 committed:
Merge pull request #5 from wj200/A-CodeQuality

Command classes: remove unused import statements

wj200 committed:
Added DoWithin class as a new type of Task

wj200 committed:
Abstract run() method and main() function of Mars class to the getResponse() method.

Added getResponse() method to UI class to facilitate the above.

wj200 committed:
Abstract printing functionality of each Command via a setter method in UI class

wj200 committed:
Fix syntax errors

wj200 committed:
Remove usage of Horizontal Lines

wj200 committed:
Fix bugs

wj200 committed:
Fix user guide formatting

wj200 committed:
Remove unused import statements

wj200 committed:
Added import statements

wj200 committed:
Merge branch 'master' into Do-Within-Task

wj200 committed:
Merge pull request #6 from wj200/Do-Within-Task

Do within task

wj200 committed:
Update README.md

wj200 committed:
Added Product Image

wj200 committed:
Merge remote-tracking branch 'origin/Do-Within-Task' into Do-Within-Task

wj200 committed:
Merge pull request #7 from wj200/Do-Within-Task

Added Ui.png

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Final JAR file created for version release

wj200 committed:
Merge pull request #8 from wj200/Do-Within-Task

Final JAR file created for version release

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Create Ui.png

wj200 committed:
Delete docs/Ui.png

wj200 committed:
Moved Ui.png to root

wj200 committed:
Merge pull request #9 from wj200/Do-Within-Task

Moved Ui.png to root

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md

wj200 committed:
Update README.md
